pkg/netns/netns.go

```diff
@@ -4,88 +4,302 @@
 package netns
 
 import (
-	"errors"
 	"fmt"
 	"os"
-	"os/exec"
+	"path"
 	"path/filepath"
+	"runtime"
 	"strings"
+	"sync"
 
-	"github.com/containernetworking/plugins/pkg/ns"
-	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+
+	// TODO consider dropping this dep entirely - it is only used here now
+	"github.com/vishvananda/netns"
 )
 
-// RemoveIfFromNetNSIfExists removes the given interface from the given
-// network namespace.
+const bindMountPath = "/run/netns" /* Bind mount path for named netns */
+
+type netNSHandle struct {
+	handle int
+	Name   string
+}
+
+type NetNsHandle interface {
+	// Returns a file descriptor representing this object's network namespace
+	Fd() int
```

Comment:
```suggestion
	FD() int
```

(https://go.dev/talks/2014/names.slide#5: `Acronyms should be all capitals, as in ServeHTTP and IDProcessor.`)